### PR TITLE
Add Cluster-wide Certificates Refresh

### PR DESCRIPTION
**Pull request by mateoflorido**
## Overview
Add a proposal for implementing a cluster-wide certificates refresh in the Canonical Kubernetes providers

### CODE REVIEW

**Review by HomayoonAlimohammadi — APPROVED** [review]
Thanks a lot @mateoflorido! Something that I wonder is that do we have a retry mechanism on the single machine certs refresh reconciler? If that's the case, then you might need to consider the problem that we already have with the in-place upgrade reconcilers. 
In short, currently the single machine in-place upgrade reconciler retries by removing the `status` annotation on a machine without removing the upgrade instructions, which in a sense reverts everything back to a state that looks like a fresh upgrade command has been triggered. This can cause some confusions on the higher level which is the `MachineDeploymentReconciler`. 
There are workarounds for that tho, and I'm currently going for the one that I consider the best (adding a new annotation).

File: docs/proposals/004-orchestration-refresh-certs.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+The status of the certificate refresh process will be shared via the
+`MachineDeployment` object in the same way as the `Cluster` controller.
+
+## ControlPlane Provider Changes
+<!--
+This section MUST mention any changes to the controlplane provider.
+-->
+
+A controller `CK8sCertificatesReconciler` will be added to the control plane
```
Comment (berkayoz) [inline docs/proposals/004-orchestration-refresh-certs.md:180]:
```suggestion
A controller `ControlPlaneCertificatesReconciler` will be added to the control plane
```
Is this a copy paste error or intentional?

**Review by berkayoz — APPROVED** [review]
LGTM, great work left a small comment.

**Review by bschimke95 — COMMENTED** [review]
LGTM, great proposal @mateoflorido

File: docs/proposals/004-orchestration-refresh-certs.md (hunk 1 above)
Comment (mateoflorido) [reply docs/proposals/004-orchestration-refresh-certs.md:180]: That was intentional, but I like your naming better. Thanks!

**Comment by mateoflorido** [comment]
> Thanks a lot @mateoflorido! Something that I wonder is that do we have a retry mechanism on the single machine certs refresh reconciler? If that's the case, then you might need to consider the problem that we already have with the in-place upgrade reconcilers. In short, currently the single machine in-place upgrade reconciler retries by removing the `status` annotation on a machine without removing the upgrade instructions, which in a sense reverts everything back to a state that looks like a fresh upgrade command has been triggered. This can cause some confusions on the higher level which is the `MachineDeploymentReconciler`. There are workarounds for that tho, and I'm currently going for the one that I consider the best (adding a new annotation).

Great, we should consider this approach when we implement the proposal. Thanks, @HomayoonAlimohammadi!

**Review by bschimke95 — APPROVED** [review]
Let's change to accepted and merge the proposal.

**Pull request merged**